src/applications/app.helpers.js

```diff
@@ -57,3 +57,13 @@ export function notSkipped(item) {
 export function toName(app) {
   return app.name;
 }
+
+export function getAppProps(app) {
+    const props = {appName: app.name};
+
+    if (app.customProps) {
+        props.customProps = app.customProps;
```

Comment:
We should decide what data type the customProps have to be. If they are always an object, one option would be to flatten all the props together instead of having a customProps object.

One option:
```js
function getAppProps(app) {
  return {
    appName: app.name,
    ...app.customProps,
  };
}
```

What are your thoughts on that? It would prevent you from having to props of the same name, but it also would allow you to "override" the default props provided by single-spa.

Reply:
On the one hand I really like the idea of being able to override the default props, but I"m not convinced the ability gives us any new functionality, but it does cost implementors the ability to have whatever name they want, because it's possible it introduces name collision.

I vote object but not flattened.

Reply:
I am split over your suggestion. I don't mind either way.

Reply:
I agree with you @TheMcMurder -- let's keep it how it is with the special "customProps" field. No code changes necessary here 👍